CMakeLists.txt

```diff
@@ -26,13 +26,56 @@ if(POLICY CMP0071)
 endif()
 
 # Set up vcpkg
-if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED VCPKG_ROOT)
-    set(VCPKG_ROOT "$ENV{VCPKG_ROOT}")
+if(WIN32 OR APPLE)
+  option(VCPKG "Use vcpkg for dependencies" ON)
+else()
+  option(VCPKG "Use vcpkg for dependencies" OFF)
 endif()
 
-if(DEFINED VCPKG_ROOT)
+if(VCPKG)
+  message(STATUS "Using dependencies from vcpkg")
+
   set(X_VCPKG_APPLOCAL_DEPS_INSTALL TRUE)
 
+  if(NOT DEFINED ENV{VCPKG_BINARY_SOURCES})
+    set(ENV{VCPKG_BINARY_SOURCES} "clear;nuget,mixxx,read;default;")
+  endif()
+
+  set(ENV{VCPKG_DISABLE_COMPILER_TRACKING} "1")
+
+  if(VCPKG_ROOT)
+    message(STATUS "Using dependencies from vcpkg repository at ${VCPKG_ROOT}")
+
+    if(NOT EXISTS "${VCPKG_ROOT}/bootstrap-vcpkg.sh")
+      message(FATAL_ERROR "${VCPKG_ROOT} is not a vcpkg Git repository.")
+    endif()
+  else()
+    message(STATUS "Using dependencies from vcpkg Git submodule")
+    set(VCPKG_ROOT "${CMAKE_SOURCE_DIR}/vcpkg")
+
+    if(NOT EXISTS "${VCPKG_ROOT}/bootstrap-vcpkg.sh")
+
+      # Make sure we have actually attempted to find Git
+      if(NOT GIT_FOUND)
+        find_package(Git)
+      endif()
+
+      if(GIT_FOUND)
```

Comment:
please consider to inverse this condition to have the error message closer to GIT_FOUND